indexer/services/roundtable/src/tasks/aggregate-trading-rewards.ts

```diff
@@ -213,23 +229,305 @@ export class AggregateTradingReward {
+    return interval.start.startOf(this.getDateTimeUnit()).toISO();
+  }
+
+  private async getNextBlock(time: IsoString): Promise<string> {
+    const blocks: BlockFromDatabase[] = await BlockTable.findAll({
+      createdOnOrAfter: time,
+      limit: 3,
+    }, []);
+
+    if (blocks.length === 0) {
+      logger.error({
+        at: 'aggregate-trading-rewards#getStartedAtHeight',
+        message: 'No blocks found after time, this should never happen',
+        time,
+      });
+      throw new Error(`No blocks found after ${time}`);
+    }
+    return blocks[0].blockHeight;
+  }
+
+  private isEndofPeriod(endTime: DateTime): boolean {
+    return endTime.startOf(this.getDateTimeUnit()).toISO() === endTime.toISO();
+  }
+
+  private async addEndedAtAndEndedAtHeightUpdates(
+    aggregationUpdateAndCreateObjects: AggregationUpdateAndCreateObjects,
+    interval: Interval,
+  ): Promise<AggregationUpdateAndCreateObjects> {
+    const startedAt: string = this.getStartedAt(interval);
+    const startedAtHeight: string = await this.getNextBlock(startedAt);
+    const endedAt: IsoString = this.getEndedAt(interval);
+    // endedAtHeight is the first block created before endedAt
+    const endedAtHeight: string = Big(await this.getNextBlock(endedAt)).minus(1).toFixed();
+
+    const allIncompleteAggregation: TradingRewardAggregationFromDatabase[] = await
+    TradingRewardAggregationTable.findAll({
+      period: this.period,
+      startedAtHeight,
+    }, []);
+    const allIncompleteAggregationAddresses: string[] = _.map(
+      allIncompleteAggregation,
+      TradingRewardAggregationColumns.address,
+    );
+
+    const otherAggregationAddresses: string[] = _.difference(
+      allIncompleteAggregationAddresses,
+      _.map(
+        aggregationUpdateAndCreateObjects.updateObjects,
+        TradingRewardAggregationColumns.address,
+      ),
+    );
+    const otherAggregationUpdateObjects: TradingRewardAggregationUpdateObject[] = _.map(
+      otherAggregationAddresses,
+      (address: string) => {
+        return {
+          id: TradingRewardAggregationTable.uuid(address, this.period, startedAtHeight),
+          endedAt,
+          endedAtHeight,
+        };
+      },
+    );
+    return {
+      createObjects: _.map(
+        aggregationUpdateAndCreateObjects.createObjects,
+        (createObject: TradingRewardAggregationCreateObject) => {
+          return {
+            ...createObject,
+            endedAt,
+            endedAtHeight,
+          };
+        },
+      ),
+      updateObjects: _.map(
+        aggregationUpdateAndCreateObjects.updateObjects,
+        (updateObject: TradingRewardAggregationUpdateObject) => {
+          return {
+            ...updateObject,
+            endedAt,
+            endedAtHeight,
+          } as TradingRewardAggregationUpdateObject;
+        },
+      ).concat(otherAggregationUpdateObjects),
+    };
+  }
+
+  private getEndedAt(interval: Interval): IsoString {
```

Comment:
add function comment

Reply:
realized I could actually just use interval.end.toISO(), deleted this method